lib/licenses/helpers.nix

```diff
@@ -0,0 +1,133 @@
+{ lib }:
+let
+  recCheck =
+    check: attrs:
+    if attrs.type == "simple" then
+      check attrs
+    else if attrs.type == "compound" then
+      if attrs.operator == "OR" then
+        recCheck check attrs.base || recCheck check attrs.mod
+      else if attrs.operator == "AND" then
+        recCheck check attrs.base && recCheck check attrs.mod
+      else if attrs.operator == "WITH" then
+        recCheck check attrs.base && recCheck check attrs.mod
+      else
+        throw "Unknown license operator"
+    else
+      throw "Unknown license type or legacy license";
+  recCheck' =
```

Comment:
Am I missing something or is this just duplicated code?

Reply:
`||` and `&&` operator swaped